x-pack/plugins/infra/public/components/navigation/routed_tabs.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLink, EuiTab, EuiTabs } from '@elastic/eui';
+import React from 'react';
+import { Route } from 'react-router-dom';
+
+import { euiStyled } from '../../../../observability/public';
+
+interface TabConfiguration {
+  title: string | React.ReactNode;
+  path: string;
+}
+
+interface RoutedTabsProps {
+  tabs: TabConfiguration[];
+}
+
+const noop = () => {};
+
+export class RoutedTabs extends React.Component<RoutedTabsProps> {
+  public render() {
+    return <EuiTabs display="condensed">{this.renderTabs()}</EuiTabs>;
+  }
+
+  private renderTabs() {
+    return this.props.tabs.map(tab => {
+      return (
+        <Route
+          key={`${tab.path}-${tab.title}`}
+          path={tab.path}
+          children={({ match, history }) => {
+            return (
+              <TabContainer className="euiTab">
+                {/* eslint-disable-next-line @elastic/eui/href-or-on-click */}
+                <EuiLink
+                  href={history.createHref({
```

Comment:
Neat, I didn't know that API existed :bulb:

Reply:
I only found out with this work, really useful 👌